migen/build/platforms/sinara/kasli_soc.py

```diff
@@ -0,0 +1,314 @@
+from migen.build.generic_platform import *
+from migen.build.xilinx import XilinxPlatform
+
+
+_io = [
+    ("clk500", 0, Pins("B24"), IOStandard("LVCMOS25")),
+
+    ("user_led", 0, Pins("AF19"), IOStandard("LVCMOS25")),
+    ("user_led", 1, Pins("AF23"), IOStandard("LVCMOS25")),
+    ("user_led", 2, Pins("A23"), IOStandard("LVCMOS25")),
+    ("error_led", 0, Pins("D20"), IOStandard("LVCMOS25")),
+
+    ("serial", 0,
+        Subsignal("tx", Pins("Y18")),
+        Subsignal("rx", Pins("AA18")),
+        IOStandard("LVCMOS25"),
+    ),
+
+    ("user_sma_clock", 0,
+        Subsignal("p", Pins("AD20"), IOStandard("LVCMOS25"),
+            Misc("DIFF_TERM=TRUE")),
+        Subsignal("n", Pins("AD21"), IOStandard("LVCMOS25"),
+            Misc("DIFF_TERM=TRUE"))
+    ),
+
+    ("i2c", 0,
+        Subsignal("scl", Pins("B22")),
+        Subsignal("sda", Pins("B20")),
+        IOStandard("LVCMOS18")
+    ),
+
+    ("clk_gtp", 0,
+        Subsignal("p", Pins("R6")),
+        Subsignal("n", Pins("R5")),
+        IOStandard("LVCMOS25")
+    ),
+
+    ("clk125_gtp", 0,
+        Subsignal("p", Pins("U6")),
+        Subsignal("n", Pins("U5")),
+        IOStandard("LVCMOS25")
+    ),
+
+    ("sfp", 0,
+        Subsignal("txp", Pins("AA2")),
+        Subsignal("txn", Pins("AA1")),
+        Subsignal("rxp", Pins("AB4")),
+        Subsignal("rxn", Pins("AB3")),
+    ),
+
+    ("sfp", 1,
+        Subsignal("txp", Pins("W2")),
+        Subsignal("txn", Pins("W1")),
+        Subsignal("rxp", Pins("Y4")),
+        Subsignal("rxn", Pins("Y3")),
+    ),
+
+    ("sfp", 2,
+        Subsignal("txp", Pins("U2")),
+        Subsignal("txn", Pins("U1")),
+        Subsignal("rxp", Pins("V4")),
+        Subsignal("rxn", Pins("V3")),
+    ),
+
+    ("sfp", 3,
+        Subsignal("txp", Pins("R2")),
+        Subsignal("txn", Pins("R1")),
+        Subsignal("rxp", Pins("T4")),
+        Subsignal("rxn", Pins("T3")),
+    ),
+
+    ("si5324_clkin", 0,
+        Subsignal("p", Pins("G7")),
+        Subsignal("n", Pins("F7")),
+        IOStandard("LVDS_18"),
+    ),
+    ("si5324_clkout", 0,
+        Subsignal("p", Pins("AD23")),
+        Subsignal("n", Pins("AD24")),
+    ),
+    ("si5324_clkout_gtp", 0,
+        Subsignal("p", Pins("R6")),
+        Subsignal("n", Pins("R5")),
```

Comment:
That's redundant with ``clk_gtp``, and ``si5324_clkout_gtp`` is obsolete on boards that have WRPLL support like this one.